nixos/modules/services/development/mlflow/default.nix

```diff
@@ -0,0 +1,137 @@
+{ options, config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.mlflow;
+
+  keyToFlag = k: (if lib.stringLength k == 1 then "-" else "--") + k;
+
+  toFlags = k: v:
+    if v == false || v == null then
+      ""
+    else if v == true then
+      keyToFlag k
+    else if lib.isList v then
+      lib.concatMapStringsSep " " (x: toFlags k x) v
+    else
+      "${keyToFlag k} ${toString v}";
+
+  flags = lib.concatStringsSep " " (lib.mapAttrsToList toFlags cfg.settings);
+
+  valueType = with lib.types;
+    nullOr (oneOf [ bool float int package path port str ]);
```

Comment:
It is surprising that `package` is an allowed type, can you write a comment about this?

Reply:
I guess this might be covered by path already so maybe it's not necessary. I was just thinking of anything you might want to pass to a flag.